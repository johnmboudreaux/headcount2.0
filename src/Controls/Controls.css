.welcome {
  font-family: "Montserrat", sans-serif;
  padding: 12px;
  font-size: 3em;
  color: #fffed5; }
