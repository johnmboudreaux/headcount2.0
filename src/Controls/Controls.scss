@import '../Styles/variables.scss';

.welcome {
  font-family: $montserrat;
  padding: 12px;
  font-size: 3em;
  color: $light-primary;
}
