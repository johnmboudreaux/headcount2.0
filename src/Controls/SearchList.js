import React from 'react';
import PropTypes from 'prop-types';

const SearchList= (props) => {
  return (
    <div>
      <ul className="search-list">
        {props.possibleMatches.map((location, index, event) => {
          return (
            <li
              href="location"
              className="list-items"
              key={index}
            >
              <button
                onClick={() => {
                  props.handleClick(location.location);
                }}
              >{location.location}
              </button>
            </li>
          );
        })}
      </ul>
    </div>
  );
};

SearchList.propTypes = {
  possibleMatches: PropTypes.arrayOf(PropTypes.object)
};

export default SearchList;
