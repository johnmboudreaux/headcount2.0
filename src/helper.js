export default class DistrictRepository {
  constructor(kinderData) {
    this.kinderData = this.formatData(kinderData);
  }

  formatData(kinderData) {
    let uniqueArray = [];
    let returnArray = [];

    kinderData.forEach((dataObj, idx) => {
      let objectLoc = dataObj.Location;

      if (uniqueArray.indexOf(objectLoc) === -1) {
        uniqueArray.push(objectLoc);
        returnArray.push(
          {"location": objectLoc.toUpperCase(), "data": {}}
        );
      }

      returnArray.forEach((returnObj) => {
        if (returnObj.location.toUpperCase()===objectLoc.toUpperCase()) {
          if (isNaN(dataObj.Data)) {
            dataObj.Data = 0;
          }
          returnObj.data[dataObj.TimeFrame]=Math.round(1000*dataObj.Data)/1000;
        }
      });
    });

    return returnArray;
  }

  findByName(inputName) {
    let returnObj;

    if (!inputName) {
      return undefined;
    }

    inputName = inputName.toUpperCase();
    this.kinderData.forEach((dataObj, idx) => {
      if (dataObj.location.toUpperCase() === inputName) {
        returnObj = dataObj;
      }
    });
    return returnObj;
  }

  findAllMatches(inputName) {
    if (inputName) {
      let returnArray = [];

      inputName = inputName.toUpperCase();
      returnArray = this.kinderData.filter((dataObj) => {
        if (dataObj.location.indexOf(inputName) !== -1) {
          return true;
        } else {
          return false;
        }
      });
      return returnArray;
    } else {
      return this.kinderData;
    }
  }

  findAverage(inputName, returnedSchool) {
    let school = this.findByName(inputName);
    let count  = 0;
    let sum    = 0;
    for (let year in school.data) {
      if (typeof year === 'string') {
        count++;
        sum += school.data[year];
      }
    }
    if (returnedSchool) {
      return {
        location: school.location,
        average: Math.round((sum / count) * 1000) / 1000
      };
    } else {
      return Math.round((sum / count) * 1000) / 1000;
    }
  }

  compareDistrictAverages(inputNameOne, inputNameTwo) {
    let schoolOne = this.findAverage(inputNameOne, true);
    let schoolTwo = this.findAverage(inputNameTwo, true);

    let returnObj = {};

    returnObj[schoolOne.location] = schoolOne.average;
    returnObj[schoolTwo.location] = schoolTwo.average;

    let comparison = schoolOne.average / schoolTwo.average;
    returnObj.compared = Math.round(comparison * 1000) / 1000;

    return returnObj;
  }
}
