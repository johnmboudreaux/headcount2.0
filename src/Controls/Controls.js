import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DataButton from '../DataButton/DataButton';
import SearchList from './SearchList';

class Controls extends Component {
  constructor() {
    super();
    this.state = {
      inputValue: "",
      possibleMatches: []
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(event) {
    let val = event.target.value;
    this.setState({
      inputValue: val,
      possibleMatches: val ? this.props.helper.findAllMatches(val) : []
    });
  }

  render() {
    return (
      <div>
        <div className="data-button-container">
          {this.props.dataArray.map((element, index)=>{
            return (
              <DataButton
                changeDataSet={this.props.changeDataSet}
                dataSet={element.districtData}
                dataName={element.name}
                key={index}
              />
            );
          })}
        </div>
        <input
          className="search-input"
          placeholder="Search for School Here"
          type="text"
          value={ this.state.inputValue }
          onChange={this.onChange}
        />
        <SearchList
          possibleMatches={this.state.possibleMatches}
          handleClick={this.props.handleClick}
        />
      </div>
    );
  }
}

Controls.propTypes = {
  dataArray: PropTypes.arrayOf(PropTypes.object),
  changeDataSet: PropTypes.func,
  handleClick: PropTypes.func,
  helper: PropTypes.object
};

export default Controls;
