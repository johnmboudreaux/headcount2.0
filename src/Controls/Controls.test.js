import ReactDom from 'react-dom';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Controls from './Controls';
import helper from '../helper';
import config from '../setupTests';
import Adapter from 'enzyme-adapter-react-15';


describe('Controls', ()=>{

  it('should render buttons for each item in the data array', () => {
    const mockFunc = jest.fn();
    const dataArray = [
      { name: "HS Graduation Rates",
        districtData: 1
      },
      { name: "Kindergartens in Full Day Program",
        districtData: 2
      },
      { name: "Grade 3 Test Data",
        districtData: 3
      }
    ];
    const controls = shallow(<Controls
      dataArray={dataArray}
      changeDataSet={mockFunc}
      onChange={mockFunc}
      helper={helper}
      handleClick={mockFunc}
    />);
    const conLength = controls.find('.data-button-container').children().length;
    expect(conLength).toEqual(3);
  });

  it('should match snapshot', () => {
    const mockFunc = jest.fn();
    const dataArray = [
      { name: "HS Graduation Rates",
        districtData: 1
      },
      { name: "Kindergartens in Full Day Program",
        districtData: 2
      },
      { name: "Grade 3 Test Data",
        districtData: 3
      }
    ];
    const controls = shallow(<Controls
      dataArray={dataArray}
      changeDataSet={mockFunc}
      onChange={mockFunc}
      helper={helper}
      handleClick={mockFunc}
    />);

    expect(controls).toMatchSnapshot();
  });
});
