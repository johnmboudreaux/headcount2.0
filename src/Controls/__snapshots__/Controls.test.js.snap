exports[`Controls should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    changeDataSet={[Function]}
    dataArray={
        Array [
            Object {
              "districtData": 1,
              "name": "HS Graduation Rates",
            },
            Object {
              "districtData": 2,
              "name": "Kindergartens in Full Day Program",
            },
            Object {
              "districtData": 3,
              "name": "Grade 3 Test Data",
            },
          ]
    }
    handleClick={[Function]}
    helper={[Function]}
    onChange={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="data-button-container">
          <DataButton
                    changeDataSet={[Function]}
                    dataName="HS Graduation Rates"
                    dataSet={1} />
          <DataButton
                    changeDataSet={[Function]}
                    dataName="Kindergartens in Full Day Program"
                    dataSet={2} />
          <DataButton
                    changeDataSet={[Function]}
                    dataName="Grade 3 Test Data"
                    dataSet={3} />
</div>,
        <p
          className="welcome">
          Click on any two districts to compare data
</p>,
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search for School Here"
          type="text"
          value="" />,
        <SearchList
          handleClick={[Function]}
          possibleMatches={Array []} />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DataButton
              changeDataSet={[Function]}
              dataName="HS Graduation Rates"
              dataSet={1} />,
            <DataButton
              changeDataSet={[Function]}
              dataName="Kindergartens in Full Day Program"
              dataSet={2} />,
            <DataButton
              changeDataSet={[Function]}
              dataName="Grade 3 Test Data"
              dataSet={3} />,
          ],
          "className": "data-button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "changeDataSet": [Function],
              "dataName": "HS Graduation Rates",
              "dataSet": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "changeDataSet": [Function],
              "dataName": "Kindergartens in Full Day Program",
              "dataSet": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "changeDataSet": [Function],
              "dataName": "Grade 3 Test Data",
              "dataSet": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Click on any two districts to compare data",
          "className": "welcome",
        },
        "ref": null,
        "rendered": "Click on any two districts to compare data",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search for School Here",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "handleClick": [Function],
          "possibleMatches": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="data-button-container">
            <DataButton
                        changeDataSet={[Function]}
                        dataName="HS Graduation Rates"
                        dataSet={1} />
            <DataButton
                        changeDataSet={[Function]}
                        dataName="Kindergartens in Full Day Program"
                        dataSet={2} />
            <DataButton
                        changeDataSet={[Function]}
                        dataName="Grade 3 Test Data"
                        dataSet={3} />
</div>,
          <p
            className="welcome">
            Click on any two districts to compare data
</p>,
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search for School Here"
            type="text"
            value="" />,
          <SearchList
            handleClick={[Function]}
            possibleMatches={Array []} />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DataButton
                changeDataSet={[Function]}
                dataName="HS Graduation Rates"
                dataSet={1} />,
              <DataButton
                changeDataSet={[Function]}
                dataName="Kindergartens in Full Day Program"
                dataSet={2} />,
              <DataButton
                changeDataSet={[Function]}
                dataName="Grade 3 Test Data"
                dataSet={3} />,
            ],
            "className": "data-button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "changeDataSet": [Function],
                "dataName": "HS Graduation Rates",
                "dataSet": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "changeDataSet": [Function],
                "dataName": "Kindergartens in Full Day Program",
                "dataSet": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "changeDataSet": [Function],
                "dataName": "Grade 3 Test Data",
                "dataSet": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Click on any two districts to compare data",
            "className": "welcome",
          },
          "ref": null,
          "rendered": "Click on any two districts to compare data",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search for School Here",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "handleClick": [Function],
            "possibleMatches": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
